Stellar-transaction.x

```diff
@@ -653,7 +676,13 @@ case ENVELOPE_TYPE_CREATE_CONTRACT_ARGS:
         Hash networkID;
         SCContractCode source;
         uint256 salt;
-    } createContractArgs;        
+    } createContractArgs;
+case ENVELOPE_TYPE_CONTRACT_AUTH:
+    struct
+    {
+        string networkPassphrase<>;
```

Comment:
wouldn't networkID suffice here ?

Reply:
network passphrase is basically the stellar equivalent of the network id, no?

Reply:
Ok, let's use network id here. I've just got an impression that passphrase is a bit more usable, but probably this doesn't matter much. FWIW this whole thing gets hashed before signing.

Reply:
It's really required only when you have more than one signer, right ? ( i.e. otherwise, it's part of the transaction rather then operation )/

Reply:
just wondering : is there such a thing like "optional field" in XDR ?
i.e. from security perspective, it sounds like it would be needed only for the case of multiple signatures or Invokes which won't be using the transaction signature.

Reply:
> It's really required only when you have more than one signer, right ?

No, the hash of this payload is what needs to be signed in every case besides invoker.

> just wondering : is there such a thing like "optional field" in XDR ?

There are optional fields (syntax is * as in C pointers). To be clear, this whole XDR message is *not* part of the transaction and is not for transfer across the network. This is the preimage of the signature payload (i.e. the account needs to sign sha256(preimage) and attach the signature(s) to the transaction).

Reply:
I see. thanks!